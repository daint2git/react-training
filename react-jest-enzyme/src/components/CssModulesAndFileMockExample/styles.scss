.root {
  background-color: white;
}

.button {
  color: black;
}