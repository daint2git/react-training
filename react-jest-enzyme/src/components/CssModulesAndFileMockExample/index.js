import React from 'react'
import styles from './styles.scss'
import file from './file.jpg'

const Example = () => (
  <div className={styles.root}>
    <button className={styles.button}>Click</button>
    <img src={file} alt="file" />
  </div>
)

export default Example
