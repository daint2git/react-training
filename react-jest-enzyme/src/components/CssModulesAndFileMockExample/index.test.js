import CssModulesAndFileMockExample from './'

describe('CssModulesAndFileMockExample', () => {
  test('1. shallow', () => {
    const wrapper = shallow(<CssModulesAndFileMockExample />)
    expect(wrapper).toMatchSnapshot()
  })

  test('2. mount', () => {
    const wrapper = mount(<CssModulesAndFileMockExample />)
    expect(wrapper).toMatchSnapshot()
  })

  test('3. render', () => {
    const wrapper = mount(<CssModulesAndFileMockExample />)
    expect(wrapper).toMatchSnapshot()
  })

  test('4. renderer', () => {
    const wrapper = renderer.create(<CssModulesAndFileMockExample />).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
