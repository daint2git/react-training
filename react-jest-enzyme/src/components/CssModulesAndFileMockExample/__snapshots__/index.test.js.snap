// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CssModulesAndFileMockExample 1. shallow 1`] = `
<div
  className="root"
>
  <button
    className="button"
  >
    Click
  </button>
  <img
    alt="file"
    src="FILE_MOCK"
  />
</div>
`;

exports[`CssModulesAndFileMockExample 2. mount 1`] = `
<div
  className="root"
>
  <button
    className="button"
  >
    Click
  </button>
  <img
    alt="file"
    src="FILE_MOCK"
  />
</div>
`;

exports[`CssModulesAndFileMockExample 3. render 1`] = `
<div
  className="root"
>
  <button
    className="button"
  >
    Click
  </button>
  <img
    alt="file"
    src="FILE_MOCK"
  />
</div>
`;

exports[`CssModulesAndFileMockExample 4. renderer 1`] = `
<div
  className="root"
>
  <button
    className="button"
  >
    Click
  </button>
  <img
    alt="file"
    src="FILE_MOCK"
  />
</div>
`;
